src/lib/viz/style/helpers/color-bins-style.ts

```diff
@@ -62,17 +63,57 @@ export function colorBinsStyle(
 
     validateParameters(opts);
 
-    if (!opts.breaks.length) {
-      const stats = meta.stats.find(f => f.name === featureProperty) as NumericFieldStats;
-      const classifier = new Classifier(stats);
-      const breaks = classifier.breaks(opts.bins - 1, opts.method);
-      return calculateWithBreaks(featureProperty, breaks, meta.geometryType, opts);
+    return calculateWithBreaks(
+      featureProperty,
+      getBreaks(opts, meta, featureProperty),
+      meta.geometryType,
+      opts
+    );
+  };
+
+  const evalFNLegend = (layer: StyledLayer, properties = {}): LegendProperties[] => {
+    const meta = layer.source.getMetadata();
+
+    if (!meta.geometryType) {
+      return [];
     }
 
-    return calculateWithBreaks(featureProperty, opts.breaks, meta.geometryType, opts);
+    const opts = defaultOptions(meta.geometryType, options);
+    const breaks = getBreaks(opts, meta, featureProperty);
+    const stats = meta.stats.find(f => f.name === featureProperty) as NumericFieldStats;
+    let ranges = [...breaks, stats.max];
+    const colors = getColors(opts.palette, ranges.length);
+    ranges = [stats.min, ...ranges];
```

Comment:
Well done ✨ 
At the beginning I have doubts, about them being 'open' <X or >X (first and last intervals), but it's nice to have here the real min and max values, so the interval is 'real'.